content/exchange/artifacts/Generic.File.LineScan.RegexHunter.yaml

```diff
@@ -36,42 +37,64 @@ parameters:
     description: "Regex pattern to detect potentially suspicious commands or payloads."
     default: "(curl|wget|bash|/tmp/|/var/tmp/|/dev/shm|base64|eval|python|perl|crontab -e|@reboot)"
 
+  - name: WhitelistPathRegex
+    type: regex
+    description: |
+      Regex pattern to exclude entire files based on path.
+      Files whose OSPath matches this regex will not be scanned.
+    default: "/etc/cron.daily/apt-compat"
+
+  - name: WhitelistLineRegex
+    type: regex
+    description: |
+      Optional regex pattern to exclude specific lines (known benign content).
+      If empty, line-level whitelisting is skipped.
+    default: ""
+
   - name: UploadMatches
     type: bool
     description: "If true, upload files that contain regex matches."
     default: false
 
 sources:
   - query: |
-      -- Step 1: Gather files from all provided globs (no symlinks)
+      -- Step 1: Gather all candidate files
       LET found_files = SELECT OSPath, Name, Size
-      FROM glob(
-        globs=PathGlobs.Glob,
-        nosymlink=true
-      )
+      FROM glob(globs=PathGlobs.Glob, nosymlink=true)
       WHERE NOT IsDir AND NOT IsLink
 
-      -- Step 2: Scan each file line by line for regex matches
+      -- Step 2: Skip files whose paths match WhitelistPathRegex
+      LET filtered_files = SELECT OSPath, Name, Size
+      FROM found_files
+      WHERE NOT OSPath =~ WhitelistPathRegex
+
+      -- Step 3: Scan each remaining file line-by-line for suspicious matches,
+      -- applying optional line-level whitelist in a single pass
       LET regex_hits = SELECT *
       FROM foreach(
-        row={ SELECT OSPath FROM found_files },
+        row={ SELECT OSPath FROM filtered_files },
         query={
           SELECT
             OSPath AS FileName,
             Line AS LineMatch
           FROM parse_lines(filename=OSPath)
           WHERE Line =~ SuspiciousRegex
+            AND if(
```

Comment:
It might be faster to just default the regex to something that just wont match then you wont need an if() here